### PR TITLE
Scoped package & Improve use with es module

### LINKED ISSUE #10: Scoped modules break when namespacing turned off
If you have a [scoped module](https://docs.npmjs.com/misc/scope), like `@myscope/mymodule` and then turn off `ember-rollup` namespacing for that module, `ember-rollup` tries to [import](https://github.com/asakusuma/ember-rollup/blob/master/src/index.js#L75) `mymodule.js` instead of the correct `@myscope/mymodule.js`.

Thanks to @gvsboy for reporting original issue.

### PR DESCRIPTION
**Pull request by xg-wang**
fix #10 #19
override #17 #18 

scoped package
---
For [scoped packages](https://docs.npmjs.com/misc/scope), import
`@<scoped>/<package>.js` instead of `<package>.js`

es module
---
- introduce option 'rollupEntry' to explicitly specify module entry for
rollup
- support both `jsnext:main` and `module` in package.json sections

misc
---
- fix mis-placed timeout expression
- add yarn.lock

### CODE REVIEW

File: src/rollup-tree.js [hunk 1]
```diff
@@ -148,7 +153,9 @@ function rollup(runtimeDependencies, transpile, addonRoot) {
       });
     }
 
-    if (moduleDir === '.') {
+    let moduleDir = path.dirname(dep.moduleName);
```
Comment (asakusuma) [inline src/rollup-tree.js]: Can this be a `const`?

File: tests/rollup-tree-test.js [hunk 2]
```diff
@@ -44,32 +44,43 @@ describe('rollup-tree', function() {
...
-                "package.json": "create",
                 "require-relative.js": "create",
                 "spaniel.js": "create"
             });
-            this.timeout(9000);
+        }));
+
+        it('rolls up a scoped module', co.wrap(function* () {
+            this.timeout(20000);
+            let dependencies = [{ fileName: '@reactivex/rxjs.js', moduleName: '@reactivex/rxjs', rollupEntry: 'dist/esm5_for_rollup/index.js' }];
```
Comment (asakusuma) [inline tests/rollup-tree-test.js]: Can this be a `const`?

File: tests/rollup-tree-test.js [hunk 3]
```diff
@@ -44,32 +44,43 @@ describe('rollup-tree', function() {
...
+            let node = rollupTree.rollup(dependencies, undefined, addonPath);
```
Comment (asakusuma) [inline tests/rollup-tree-test.js]: Can this be a `const`?

File: tests/rollup-tree-test.js [hunk 4]
```diff
@@ -44,32 +44,43 @@ describe('rollup-tree', function() {
...
-                "README.md": "create",
-                "package.json": "create",
                 "require-relative.js": "create",
                 "spaniel.js": "create"
             });
-            this.timeout(9000);
+        }));
+
+        it('rolls up a scoped module', co.wrap(function* () {
+            this.timeout(20000);
```
Comment (asakusuma) [inline tests/rollup-tree-test.js]: Does the timeout really need to be this big?

File: README.md [hunk 5]
```diff
@@ -41,6 +41,31 @@ import RSVP from 'my-addon/rsvp';
 ```
 
 
+## rollup entry
```
Comment (asakusuma) [inline README.md]: Perhaps name this `Custom Rollup Entry`? It should at least be capitalized.

File: README.md [hunk 6]
```diff
@@ -41,6 +41,31 @@ import RSVP from 'my-addon/rsvp';
...
+
+Some packages don't specify `jsnext:main` or `module` in their `package.json` but in fact provide such stuff, you can do it yourself by using `rollupEntry`.
+
+```JavaScript
+//my-addon/index.js
+var emberRollup = require('ember-rollup');
```
Comment (asakusuma) [inline README.md]: Can this be a `const`?

File: README.md [hunk 7]
```diff
@@ -41,6 +41,31 @@ import RSVP from 'my-addon/rsvp';
...
+var runtimeDependencies = [{
```
Comment (asakusuma) [inline README.md]: Can this be a `const`?

File: README.md [hunk 8]
```diff
@@ -41,6 +41,31 @@ import RSVP from 'my-addon/rsvp';
 ```
 
 
+## rollup entry
+
+Some packages don't specify `jsnext:main` or `module` in their `package.json` but in fact provide such stuff, you can do it yourself by using `rollupEntry`.
```
Comment (asakusuma) [inline README.md]: > but in fact provide such stuff, you can do it yourself by using `rollupEntry`.

perhaps change this to something like

> even though a module exists. You can manually point rollup to the module entry point file path using `rollupEntry`.

**Comment by asakusuma** [comment]
@xg-wang also, perhaps we should add a test for scoped modules but without using `rollupEntry` and a non-scoped module with `rollupEntry`? Just to cover every permutation of the new features

**Comment by stefanpenner** [comment]
In general I reserve major bumps for breaking changes and minor for additive features, patch for fixes that are neither features nor breaking changes.

So if this merely adds a feature, with no reasonable expected breakage. Minor seems fine. If it is unsafe to upgrade, major is appropriate

**Comment by xg-wang** [comment]
@asakusuma This should be ready, only need to fix CI https://github.com/asakusuma/ember-rollup/issues/34

**Comment by asakusuma** [comment]
@xg-wang there have been some recent changes so I think you still need to rebase

**Comment by xg-wang** [comment]
@asakusuma Yes I just did that

File: package.json [hunk 9]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "ember-rollup",
   "description": "Use rollup to add runtime dependencies to an ember-cli addon.",
-  "version": "1.1.1",
+  "version": "1.1.2",
```
Comment (asakusuma) [inline package.json]: I think this should be a minor bump and not a patch bump

File: README.md [hunk 10]
```diff
@@ -41,22 +41,48 @@ import RSVP from 'my-addon/rsvp';
 ```
 
 
+## Custom Rollup Entry
+
+Some packages don't specify `jsnext:main` or `module` in their `package.json` but even though
+a module exists. You can manually point rollup to the module entry point file path using rollupEntry.
```
Comment (asakusuma) [inline README.md]: Can you put ticks around rollupEntry so it becomes `rollupEntry`?

File: README.md [hunk 11]
```diff
@@ -41,22 +41,48 @@ import RSVP from 'my-addon/rsvp';
 ```
 
 
+## Custom Rollup Entry
+
+Some packages don't specify `jsnext:main` or `module` in their `package.json` but even though
```
Comment (asakusuma) [inline README.md]: The "but" is superfluous, I think it can be removed

**Pull request merged**